### PR TITLE
Integration Tests on vanity method

### PR DESCRIPTION
**Pull request by M3rs**
Refactor the command line args out of vanity method and added as parameters (to allow for testing).

Add some basic integration tests on vanity method.

Clean up mocks in tests.py (use setUp/tearDown for logger mock)

Misc pep8 cleanup

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -239,5 +249,58 @@ def test_empty(self):
...
+    def tearDown(self):
+        vanity.logger = self.old_logger
+
+    @mock.patch('vanity.normalize', side_effect=mock_normalize)
+    @mock.patch('vanity.get_json_from_url', side_effect=mock_json_data)
+    def test_vanity_count_downloads(self, mock_json_func, mock_norm_func):
+        # NB: mocks are passed in reverse order
+        expected_url = 'https://pypi.python.org/pypi/fake package/json'
+
+        vanity.vanity(packages=['fake package'],
```
Comment (hugovk) [inline tests.py]: Likewise about package name

File: tests.py [hunk 2]
```diff
@@ -239,5 +249,58 @@ def test_empty(self):
...
+        self.mock_logger.debug.assert_any_call(
+            '%s has been downloaded %s times!',
+            'fake package',
+            '6')
+
+    @mock.patch('vanity.normalize', side_effect=mock_normalize)
+    @mock.patch('vanity.get_json_from_url', side_effect=mock_json_data)
+    def test_vanity_version_downloads(self, mock_json_func, mock_norm_func):
+        # NB: mocks are passed in reverse order
+        expected_url = 'https://pypi.python.org/pypi/fake package/json'
```
Comment (hugovk) [inline tests.py]: Ditto

**Comment by M3rs** [comment]
Not sure why this is failing code-quality, following the link it shows no errors. 

Nothing shows up from "pep8" either.

**Comment by hugovk** [comment]
Thanks!

**Pull request merged**